### DISCUSSION TITLE
issue: Render function from Controller doesn't react on the change in form values.

### DISCUSSION THREAD

**Discussion by VasiliPodoplielov** [react-hook-form/react-hook-form#12266] [Q&A]
### Version Number

7.51.5

### Codesandbox/Expo snack

https://stackblitz.com/edit/qkhbq8?file=src%2FApp.jsx

### Steps to reproduce

1. Change the value in the dropdown.
2. Click 'toggle' button to hide the form.
3. Click 'toggle' button to show the form.
4. Click on the clear icon in the Dropdown.
5. There is no console.log but the value in the form is undefined.

### Expected behaviour

When user click on "clear icon" the value in the input should be epmty.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Gopakuma** [comment]
hi, 
Controller Doesn't Update field.value: While field.onChange updates the form state, it might not directly update the field.value object returned by Controller.
Use watch Hook, it lets you track changes in specific form fields. You can use it within the render function to re-render whenever the city field changes

**Comment by VasiliPodoplielov** [comment]
Looks like it's a bug. Because in others cases Controller does update the field.value.

**Comment by Gopakuma** [comment]
I would like to fix this issue, 
Can you please guide me?

**Comment by VasiliPodoplielov** [comment]
@Gopakuma From my opinion render function should react on each change in the state that is related to the current field.

**Comment by bluebill1049** [comment]
field.onChange is the one which updates form values, that clear button would need to attach the onChnage method.

**Reply by VasiliPodoplielov** [reply]
@bluebill1049 Yes. When user presses 'clear button' the onChange function invokes, form.getValues() return correct form state, but render function doesn't render new state.

**Comment by VasiliPodoplielov** [comment]
@bluebill1049 @Gopakuma Hi guys, any updates?
